### PR TITLE
feat(blog): icon to share article by url

### PR DESCRIPTION
**Pull request by majahendzel-va**

### CODE REVIEW

File: libs/blog/articles/feature-article/src/lib/article-share-icons/article-share-icons.component.ts [hunk 1]
```diff
@@ -31,6 +48,24 @@ type ShareItem = {
...
+      <a
+        #copyUrlIcon
+        role="button"
+        [attr.aria-label]="t('articleShareIcons.urlAriaLabel')"
+        [cdkCopyToClipboard]="window.location.href"
+        (click)="onCopyUrl()"
+        target="_blank"
+      >
+        @if (!isCopyUrlClicked()) {
+          <fast-svg class="text-al-foreground" [name]="'link'" size="28" />
```
Comment (DamianBrzezinskiHoA) [inline libs/blog/articles/feature-article/src/lib/article-share-icons/article-share-icons.component.ts:60]: My proposition:
`name="link"`

File: libs/blog/articles/feature-article/src/lib/article-share-icons/article-share-icons.component.ts [hunk 2]
```diff
@@ -64,4 +105,27 @@ export class ArticleShareIconsComponent {
       },
     ];
   });
+
+  onCopyUrl(): void {
+    this.isCopyUrlClicked.set(true);
+
+    if (this.copyUrlIcon()) {
```
Comment (DamianBrzezinskiHoA) [inline libs/blog/articles/feature-article/src/lib/article-share-icons/article-share-icons.component.ts:112]: Maybe like this:
` if (!this.copyUrlIcon()) return `

File: libs/blog/articles/feature-article/src/lib/article-share-icons/article-share-icons.component.ts [hunk 3]
```diff
@@ -39,6 +74,12 @@ export class ArticleShareIconsComponent {
   title = input.required<string>();
   language = input.required<string>();
 
+  isCopyUrlClicked = signal(false);
+  copyUrlIcon = viewChild<ElementRef>('copyUrlIcon');
+
+  protected readonly window = window;
+  private renderer = inject(Renderer2);
```
Comment (mateuszbasinski) [inline libs/blog/articles/feature-article/src/lib/article-share-icons/article-share-icons.component.ts:81]:
```suggestion
  private readonly _renderer = inject(Renderer2);
```

**Review by DamianBrzezinskiHoA — APPROVED** [review]

File: libs/blog/articles/feature-article/src/lib/article-share-icons/article-share-icons.component.ts [hunk 4]
```diff
@@ -31,6 +48,20 @@ type ShareItem = {
...
+        [attr.aria-label]="t('articleShareIcons.urlAriaLabel')"
+        [cdkCopyToClipboard]="window.location.href"
+        (click)="onCopyUrl()"
+        target="_blank"
+      >
+        @if (!isCopyUrlClicked()) {
+          <fast-svg name="link" class="text-al-foreground" size="28" />
+        } @else {
+          <fast-svg name="circle-check" class="text-al-foreground" size="28" />
+        }
```
[the remark this answers is not in the document]
Comment (DDonochVA) [reply libs/blog/articles/feature-article/src/lib/article-share-icons/article-share-icons.component.ts:63]: nvm, it looks like it doesn't handle it

File: libs/blog/articles/feature-article/src/lib/article-share-icons/article-share-icons.component.ts [hunk 5]
```diff
@@ -31,6 +48,20 @@ type ShareItem = {
...
+        role="button"
+        [attr.aria-label]="t('articleShareIcons.urlAriaLabel')"
+        [cdkCopyToClipboard]="window.location.href"
+        (click)="onCopyUrl()"
+        target="_blank"
+      >
+        @if (!isCopyUrlClicked()) {
+          <fast-svg name="link" class="text-al-foreground" size="28" />
+        } @else {
+          <fast-svg name="circle-check" class="text-al-foreground" size="28" />
```
Comment (DDonochVA) [inline libs/blog/articles/feature-article/src/lib/article-share-icons/article-share-icons.component.ts:62]: there will be a frame where unloaded icons shows up on the screen as there are some optimizations in this component, so it's better to load them both and use classes to hide one or another

File: libs/blog/articles/feature-article/src/lib/article-share-icons/article-share-icons.component.ts [hunk 6]
```diff
@@ -13,16 +24,22 @@ type ShareItem = {
 @Component({
   standalone: true,
   selector: 'al-article-share-icons',
-  imports: [IconComponent, TranslocoDirective, FastSvgComponent],
+  imports: [
+    IconComponent,
+    TranslocoDirective,
+    FastSvgComponent,
+    CdkCopyToClipboard,
+    NgClass,
```
Comment (DDonochVA) [inline libs/blog/articles/feature-article/src/lib/article-share-icons/article-share-icons.component.ts:32]: is this used?

**Review by DDonochVA — COMMENTED** [review]
try this
- no viewchild
- no renderer
- no window
- less code

```ts
import { CdkCopyToClipboard } from '@angular/cdk/clipboard';
import {
  Component,
  computed,
  input,
  signal,
} from '@angular/core';
import { TranslocoDirective } from '@jsverse/transloco';
import { FastSvgComponent } from '@push-based/ngx-fast-svg';

type ShareItem = {
  href: string;
  icon: string;
  ariaLabel: string;
};

@Component({
  standalone: true,
  selector: 'al-article-share-icons',
  imports: [
    TranslocoDirective,
    FastSvgComponent,
    CdkCopyToClipboard,
  ],
  styleUrls: ['./article-share-icons.component.scss'],
  template: `
    <div *transloco="let t" class="flex items-center gap-4">
      <span class="text-lg font-bold">
        {{ t('articleShareIcons.title') }}
      </span>

      @for (item of items(); track $index) {
        <a
          role="button"
          target="_blank"
          [attr.aria-label]="t(item.ariaLabel)"
          [href]="item.href"
        >
          <fast-svg class="text-al-foreground" [name]="item.icon" size="28" />
        </a>
      }
      <a
        role="button"
        target="_blank"
        [attr.aria-label]="t('articleShareIcons.urlAriaLabel')"
        [class.url-icon-animated]="animating()"
        [cdkCopyToClipboard]="articleUrl()"
        (animationend)="animating.set(false)"
        (click)="animating.set(true)"
      >
        <fast-svg [class.!hidden]="animating()" name="link" class="text-al-foreground" size="28" />
        <fast-svg [class.!hidden]="!animating()" name="circle-check" class="text-al-foreground" size="28" />
      </a>
    </div>
  `,
})
export class ArticleShareIconsComponent {
  readonly slug = input.required<string>();
  readonly title = input.required<string>();
  readonly language = input.required<string>();

  readonly animating = signal(false);

  readonly articleUrl = computed(() => this.language() === 'pl_PL'
    ? `https://angular.love/${this.slug()}`
    : `https://angular.love/en/${this.slug()}`);

  readonly items = computed<ShareItem[]>(() => {
    const text = encodeURIComponent(this.title());
    const url = this.articleUrl();

    return [
      {
        icon: 'twitter-x',
        href: `https://x.com/intent/tweet?text=${text}&url=${url}&hashtags=angularlove`,
        ariaLabel: 'articleShareIcons.twitterAriaLabel',
      },
      {
        icon: 'linkedIn',
        href: `https://www.linkedin.com/shareArticle?mini=true&url=${url}`,
        ariaLabel: 'articleShareIcons.linkedInAriaLabel',
      },
      {
        icon: 'facebook',
        href: `https://www.facebook.com/sharer/sharer.php?u=${url}`,
        ariaLabel: 'articleShareIcons.facebookAriaLabel',
      },
    ];
  });
}
```

File: libs/blog/articles/feature-article/src/lib/article-share-icons/article-share-icons.component.ts [hunk 7]
```diff
@@ -13,16 +14,21 @@ type ShareItem = {
 @Component({
   standalone: true,
   selector: 'al-article-share-icons',
-  imports: [IconComponent, TranslocoDirective, FastSvgComponent],
+  imports: [
+    IconComponent,
```
Comment (DDonochVA) [inline libs/blog/articles/feature-article/src/lib/article-share-icons/article-share-icons.component.ts:18]: this can be removed

**Review by DDonochVA — APPROVED** [review]

**Pull request merged**